tracing/tests/support/subscriber.rs

```diff
@@ -81,6 +81,7 @@ where
         self
     }
 
+    #[allow(deprecated)]
```

Comment:
Why isn't this like the others?

Reply:
This is allowed because I still want the tests to be able to assert that `drop_span` gets called. Eventually, we should add the ability to expect a `try_close` call explicitly to the test subscriber, and update the tests to expect that instead; since this branch touches a lot of code, I opted to make the smallest possible changes here & make bigger ones in more focused follow-up branches.